clients/go/coreutils/literals.go

```diff
@@ -492,7 +493,9 @@ func MakeLiteralForType(t *core.LiteralType, v interface{}) (*core.Literal, erro
 		if v == nil {
 			strValue = ""
 		}
-
+		if f, ok := v.(float64); ok && math.Trunc(f) == f {
```

Comment:
mind adding a comment here why this is necessary?

Reply:
Done.